This patch is taken from gentoo musl overlay
sigval_t is glibc only construct, we use a union of sigval
which pretty much is same effect as sigval_t

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>

diff -Naur DirectFB-1.7.6.orig/lib/direct/os/linux/glibc/system.c DirectFB-1.7.6/lib/direct/os/linux/glibc/system.c
--- DirectFB-1.7.6.orig/lib/direct/os/linux/glibc/system.c	2014-07-15 02:54:58.000000000 -0400
+++ DirectFB-1.7.6/lib/direct/os/linux/glibc/system.c	2015-07-18 16:55:35.077989166 -0400
@@ -111,7 +111,7 @@
 void
 direct_trap( const char *domain, int sig )
 {
-     sigval_t val;
+     union sigval val;
 
      if (direct_config->delay_trap_ms) {
           D_LOG( Direct_Trap, VERBOSE, "NOT RAISING signal %d from %s, waiting for %dms... attach gdb --pid=%d\n", sig, domain, direct_config->delay_trap_ms, getpid() );
